common/lib/dependabot/clients/azure.rb

```diff
@@ -0,0 +1,214 @@
+          "/_apis/git/repositories/" + source.unscoped_repo)
+
+        JSON.parse(response.body).fetch("defaultBranch").gsub("refs/heads/", "")
+      end
+
+      def fetch_repo_contents(commit = nil, path = nil)
+        tree = fetch_repo_contents_treeroot(commit, path)
+
+        response = get(source.api_endpoint +
+          source.organization + "/" + source.project +
+          "/_apis/git/repositories/" + source.unscoped_repo +
+          "/trees/" + tree + "?recursive=false")
+
+        JSON.parse(response.body).fetch("treeEntries")
+      end
+
+      def fetch_repo_contents_treeroot(commit = nil, path = nil)
+        actual_path = path
+        actual_path = "/" if path.to_s.empty?
+
+        tree_url = source.api_endpoint +
+                   source.organization + "/" + source.project +
+                   "/_apis/git/repositories/" + source.unscoped_repo +
+                   "/items?path=" + actual_path
+
+        unless commit.to_s.empty?
+          tree_url += "&versionDescriptor.versionType=commit" \
+                      "&versionDescriptor.version=" + commit
+        end
+
+        tree_response = get(tree_url)
+
+        JSON.parse(tree_response.body).fetch("objectId")
+      end
+
+      def fetch_file_contents(commit, path)
+        response = get(source.api_endpoint +
+          source.organization + "/" + source.project +
+          "/_apis/git/repositories/" + source.unscoped_repo +
+          "/items?path=" + path +
+          "&versionDescriptor.versionType=commit" \
+          "&versionDescriptor.version=" + commit)
+
+        response.body
+      end
+
+      def commits(branch_name = nil)
+        commits_url = source.api_endpoint +
+                      source.organization + "/" + source.project +
+                      "/_apis/git/repositories/" + source.unscoped_repo +
+                      "/commits"
+
+        unless branch_name.to_s.empty?
+          commits_url += "?searchCriteria.itemVersion.version=" + branch_name
+        end
+
+        response = get(commits_url)
+
+        JSON.parse(response.body).fetch("value")
+      end
+
+      def branch(branch_name)
+        response = get(source.api_endpoint +
+          source.organization + "/" + source.project +
+          "/_apis/git/repositories/" + source.unscoped_repo +
+          "/refs?filter=heads/" + branch_name)
+
+        JSON.parse(response.body).fetch("value").first
+      end
+
+      def pull_requests(source_branch, target_branch)
+        response = get(source.api_endpoint +
+          source.organization + "/" + source.project +
+          "/_apis/git/repositories/" + source.unscoped_repo +
+          "/pullrequests?searchCriteria.status=all" \
+          "&searchCriteria.sourceRefName=refs/heads/" + source_branch +
+          "&searchCriteria.targetRefName=refs/heads/" + target_branch)
+
+        JSON.parse(response.body).fetch("value").
+          reject { |pull_request| pull_request.fetch("status") == "abandoned" }
```

Comment:
Not sure. Does "abandoned" mean closed here? This method is used to check whether a PR already exists so that it isn't duplicate created. My hunch is that `abandoned` **shouldn't** be rejected here, and shouldn't be rejected in the pull request creator either.

Reply:
abandoned" does mean close from GitHub terminology, I'll remove it then. Otherwise, if someone doesn't want a dependency update and abandons it, it'll get recreated on the next dependabot run.